### PR TITLE
(PA-3974) Increase DH key size for pcp-broker

### PR DESCRIPTION
**Pull request by GabrielNagy**
Add the 'jdk.tls.ephemeralDHKeySize=2048' JVM option when running pcp-broker, as the default DH key size is too small for openssl 1.1.1k FIPS.

### CODE REVIEW

**Review by McdonaldSeanp — APPROVED** [review]

File: acceptance/lib/pxp-agent/test_helper.rb [hunk 1]
```diff
@@ -68,7 +72,8 @@ def run_pcp_broker(host, instance=0)
 
   lein_command = "cd #{GIT_CLONE_FOLDER}/pcp-broker#{instance}; export LEIN_ROOT=ok; \
      lein with-profile #{LEIN_PROFILE} tk </dev/null >/var/log/puppetlabs/pcp-broker.log.#{SecureRandom.uuid} 2>&1 &"
-  lein_command = append_jvm_limits(lein_command) if host[:gke_container]
+  lein_command = append_jvm_opts('-Xms2g -Xmx2g', lein_command) if host[:gke_container]
```
Comment (donoghuc) [inline acceptance/lib/pxp-agent/test_helper.rb:75]: Is this method used anywhere else? Breaking this in to a method that does some complicated string manipulation makes this pretty hard to reason about. It looks like the only special case is if the host is a GKE cluster right? So could we just do something simple like:

```ruby
jvm_opts = ['-Djdk.tls.ephemeralDHKeySize=2048']
jvm_opts << '-Xms2g -Xmx2g' if host[:gke_container]
lein_command = "export JVM_OPTS='#{jvm_opts.join(' ')}'; export LEIN_JVM_OPTS='#{jvm_opts.join(' ')}'; #{lein_command}"
'''
```

File: acceptance/setup/common/050_Setup_Broker.rb [hunk 2]
```diff
@@ -55,7 +59,7 @@
   # some time and will eat into the polling period we allow for the broker to start
   for i in 0..NUM_BROKERS-1
     lein_command = "cd #{GIT_CLONE_FOLDER}/pcp-broker#{i}; export LEIN_ROOT=ok; lein with-profile #{LEIN_PROFILE} deps"
-    lein_command = append_jvm_limits(lein_command) if master[:gke_container]
+    lein_command = append_jvm_opts('-Xms2g -Xmx2g', lein_command) if master[:gke_container]
```
Comment (donoghuc) [inline acceptance/setup/common/050_Setup_Broker.rb:62]: ah, i see this is where that method is used. I think its simple enough just to conditionally append it here without using a general method https://github.com/puppetlabs/pxp-agent/pull/835/files#r701994227

**Comment by donoghuc** [comment]
Thanks so much @GabrielNagy for tracking this down! Excellent work.

**Comment by steveax** [comment]
Yes, thank you! 🙏

**Pull request merged**